manifests/charts/istio-control/istio-discovery/files/injection-template.yaml

```diff
@@ -467,6 +471,12 @@ spec:
     configMap:
       name: istio-ca-root-cert
   {{- end }}
+  {{- if eq .Values.global.pilotCertProvider "kubernetes" }}
+  - name: kube-ca-cert
```

Comment:
Do we want another one? Or make this `istiod-ca-cert` as well, mount to the same place, and then no code changes needed?

Reply:
In the code when pilotCertProvider is kubernetes,  to have no code change I think we need to mount to `/var/run/secrets/kubernetes.io/serviceaccount`. 
We could also make this `istiod-ca-cert` as you said, make it not optional and then change the k8sCAPath same as CitadelCACertPath i.e `./var/run/secrets/istio`. This could also mean we can skip the `else if a.secOpts.PilotCertProvider == constants.CertProviderKubernetes` condition entirely. This would decrease the changes but I think it would not be explicit in this part of the code which cert was used.

Reply:
are the tests in tests/integration/security/mtlsk8sca/ still running in presubmit? ability to use K8sCA doesn't exist for cluster >= 1.22.

Reply:
would it also make sense in our integration tests to deploy echo workloads with service accounts where autoMountServiceToken is set to false ? Perhaps we can make this the default configuration for test echo workloads..

Reply:
That was my initial idea but then I got worried we would somehow miss some bug when autoMountServiceToken=true.. so it seemed reasonable to just set it for some random workload that way we have coverage for both